presto/presto_test.go

```diff
@@ -191,6 +191,79 @@ func TestRoundTripCancellation(t *testing.T) {
 	}
 }
 
+func TestQueryContextCancellation(t *testing.T) {
```

Comment:
Can we test it as an integration test and run queries against actual Trino instance run in docker?. 
Testing against mock HTTP server works but is very verbose and simulates specific server behavior some of which may be treated as an implementation detail.

We should be able to schedule a real long running query (e.g. `SELECT COUNT(*) from tpch.sf10.lineitem`. Cancel it from sub routine. And then verify with another query that original query was cancelled on the server side by using `system.runtime.queries` table (while query is still running you can determine its query id by querying same table).

Reply:
I replaced this test with an integration one. While writing the test I found multiple issues so please review the whole PR again.